### ISSUE TITLE
NPE while InventoryOpenEvent

### ISSUE THREAD

**Issue by Schneidertm** [redstonefreak589/Honeypot#8]
Hey, thank you for this project and for maintaining it! While running this on my little server I encounter this NPE. While its nothing game breaking I thought I still letting you know.

```
[01:22:24] [Server thread/ERROR]: Could not pass event InventoryOpenEvent to Honeypot v1.4.1
java.lang.NullPointerException: null
        at java.util.Objects.requireNonNull(Unknown Source) ~[?:?]
        at me.terrorbyte.honeypot.events.PlayerContainerOpenListener.InventoryOpenEvent(PlayerContainerOpenListener.java:24) ~[honeypot-1.4.1.jar:?]
        at com.destroystokyo.paper.event.executor.StaticMethodHandleEventExecutor.execute(StaticMethodHandleEventExecutor.java:38) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:git-Paper-290]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:629) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_18_R2.event.CraftEventFactory.callInventoryOpenEvent(CraftEventFactory.java:1355) ~[paper-1.18.2.jar:git-Paper-290]
        at org.bukkit.craftbukkit.v1_18_R2.event.CraftEventFactory.callInventoryOpenEvent(CraftEventFactory.java:1341) ~[paper-1.18.2.jar:git-Paper-290]
        at net.minecraft.server.level.ServerPlayer.openHorseInventory(ServerPlayer.java:1539) ~[?:?]
        at net.minecraft.world.entity.animal.horse.AbstractHorse.openInventory(AbstractHorse.java:430) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.handlePlayerCommand(ServerGamePacketListenerImpl.java:2429) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundPlayerCommandPacket.handle(ServerboundPlayerCommandPacket.java:37) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundPlayerCommandPacket.handle(ServerboundPlayerCommandPacket.java:7) ~[?:?]
        at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:51) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.18.2.jar:git-Paper-290]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[?:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1399) ~[paper-1.18.2.jar:git-Paper-290]
        at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:188) ~[paper-1.18.2.jar:git-Paper-290]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1376) ~[paper-1.18.2.jar:git-Paper-290]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1369) ~[paper-1.18.2.jar:git-Paper-290]
        at net.minecraft.util.thread.BlockableEventLoop.runAllTasks(BlockableEventLoop.java:114) ~[?:?]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1503) ~[paper-1.18.2.jar:git-Paper-290]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1225) ~[paper-1.18.2.jar:git-Paper-290]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:315) ~[paper-1.18.2.jar:git-Paper-290]
        at java.lang.Thread.run(Unknown Source) ~[?:?]
```
```
[00:18:33] [Server thread/ERROR]: Could not pass event InventoryOpenEvent to Honeypot v1.4.1
java.lang.NullPointerException: null
	at java.util.Objects.requireNonNull(Unknown Source) ~[?:?]
	at me.terrorbyte.honeypot.events.PlayerContainerOpenListener.InventoryOpenEvent(PlayerContainerOpenListener.java:24) ~[honeypot-1.4.1.jar:?]
	at com.destroystokyo.paper.event.executor.StaticMethodHandleEventExecutor.execute(StaticMethodHandleEventExecutor.java:38) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:git-Paper-287]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:629) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_18_R2.event.CraftEventFactory.callInventoryOpenEvent(CraftEventFactory.java:1355) ~[paper-1.18.2.jar:git-Paper-287]
	at net.minecraft.server.level.ServerPlayer.openMenu(ServerPlayer.java:1493) ~[?:?]
	at net.minecraft.world.item.trading.Merchant.openTradingScreen(Merchant.java:41) ~[?:?]
	at net.minecraft.world.entity.npc.WanderingTrader.mobInteract(WanderingTrader.java:120) ~[?:?]
	at net.minecraft.world.entity.Mob.interact(Mob.java:1258) ~[?:?]
	at net.minecraft.world.entity.player.Player.interactOn(Player.java:1097) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl$5.performInteraction(ServerGamePacketListenerImpl.java:2506) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl$5.a(ServerGamePacketListenerImpl.java:2525) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundInteractPacket$InteractionAction.dispatch(ServerboundInteractPacket.java:131) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundInteractPacket.dispatch(ServerboundInteractPacket.java:80) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleInteract(ServerGamePacketListenerImpl.java:2467) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundInteractPacket.handle(ServerboundInteractPacket.java:67) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundInteractPacket.handle(ServerboundInteractPacket.java:12) ~[?:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:51) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.18.2.jar:git-Paper-287]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1399) ~[paper-1.18.2.jar:git-Paper-287]
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:188) ~[paper-1.18.2.jar:git-Paper-287]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1376) ~[paper-1.18.2.jar:git-Paper-287]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1369) ~[paper-1.18.2.jar:git-Paper-287]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1347) ~[paper-1.18.2.jar:git-Paper-287]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1229) ~[paper-1.18.2.jar:git-Paper-287]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:315) ~[paper-1.18.2.jar:git-Paper-287]
	at java.lang.Thread.run(Unknown Source) ~[?:?]
```

My environment: 
- Docker container (itzg/minecraft-server)
- Java 17
- Paper 1.18.2 (287 & 290)
- Honeypot 1.4.1

**Comment by redstonefreak589** [comment]
Hey there! Thanks for the info, I'll take a look and post an update. I've been meaning to work on this project anyway! Looks like these were caused by opening non-standard inventories (In this case, Horse and Wandering Trader). I'll add some checks to the code and get an update pushed to prevent these NPEs 😄

**Comment by redstonefreak589** [comment]
Okay, I found the problem and am implementing a fix. In my logic for "PlayerContainerOpenListener" I assumed that all inventories the player will open will be a block. I did not account for entity based inventories like Horses, Villagers, etc., which cause an NPE because it's immediately checking what block the player was targeting during the inventory open event (which will be null since the player isn't opening a block-based inventory). Thanks for your help in finding this bug! I'll be pushing 1.4.2 within the next 24 hours, and I'm adding a feature request to allow inventory filtering, which should be a relatively easy feature to add 😄

**Comment by redstonefreak589** [comment]
Resolved in version v1.4.2. Added a try-catch block to prevent NPEs from occuring.

**Comment by redstonefreak589** [comment]
Totally screwed this up, I accidentally inverted an if statement in the code that will basically prevent inventory actions from activating. However, [I just posted v1.5](https://github.com/redstonefreak589/Honeypot/releases/tag/v1.5) which adds some pretty sweet improvements, and also fixes this bug! I'd recommend downloading it ASAP 😉
